pets/peer-finder/peer-finder.go

```diff
@@ -74,17 +75,42 @@ func main() {
 	if ns == "" {
 		ns = os.Getenv("POD_NAMESPACE")
 	}
-	if *svc == "" || ns == "" || (*onChange == "" && *onStart == "") {
-		log.Fatalf("Incomplete args, require -on-change and/or -on-start, -service and -ns or an env var for POD_NAMESPACE.")
-	}
-
 	hostname, err := os.Hostname()
 	if err != nil {
 		log.Fatalf("Failed to get hostname: %s", err)
 	}
+	var domainName string
+
+	// If domain is not provided, try to get it from resolv.conf
+	if (*domain == "") || (ns == "") {
+		resolvConfBytes, err := ioutil.ReadFile("/etc/resolv.conf")
+		resolvConf := string(resolvConfBytes)
+		if err != nil {
+			log.Fatal("Unable to read /etc/resolv.conf")
+		}
+		if ns != "" {
+			tempResolv := strings.Split(resolvConf, ("search " + ns + "."))
```

Comment:
If the `resolv.conf` does not contain the cluster domain, I believe the dns e2e conformance test will fail - so I believe we can expect this behavior.  From a user perspective, there is a presumption in kubernetes that you can skip the FQDN and use the shorter domain name, and many scripts rely on that.  That only works if there is a search line.

I could be wrong but have you ever seen resolv.conf _not_ have the pod's domain in it?

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/dns.go shows that a conformance test searches for `kubernetes`, `kubernetes.default`, `kubernetes.default.svc`, and then finally the FQDN.